### PR TITLE
[4.0] remove redundant code & general cleanup

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

remove redundant code (and a todo) 

`$database` is not used 

added a `\` to a fqdn to allow for IDE auto completion and cmd+click

Convert `array` to `[]` syntax

### Testing Instructions

Turn on the debug mode to get the pgpDebugBar at the bottom - open it up to the queries page

### Actual result BEFORE applying this Pull Request

Everything is logged fine

### Expected result AFTER applying this Pull Request

Everything is logged fine - no change. 

IDE can hover cmd+click full class 

Screen Recording 2020-06-21 at 09 54 PM

### Documentation Changes Required

None.

### CODE REVIEW

File: plugins/system/debug/src/DataCollector/QueryCollector.php [hunk 1]
```diff
@@ -100,9 +100,6 @@ public function __construct(Registry $params, DebugMonitor $queryMonitor, array
 	 */
 	public function collect(): array
 	{
-		// @todo fetch the database object in a non deprecated way..
-		$database = Factory::$database;
```
Comment (Quy) [inline plugins/system/debug/src/DataCollector/QueryCollector.php:104]: Please remove `use Joomla\CMS\Factory;`

Comment (PhilETaylor) [reply plugins/system/debug/src/DataCollector/QueryCollector.php:104]: Mouse battery is dead - in bed - will do tomorrow

Comment (PhilETaylor) [reply plugins/system/debug/src/DataCollector/QueryCollector.php:104]: Done.

**Comment by wilsonge** [comment]
Merged on code review. One thing to note. As a general rule for future things can we leave out the array to `[]` conversions in - it causes more merge conflicts than it's worth when trying to merge up 3.x into 4.x - it's something we should probably do after 3.x goes EOL for maintainability (In this specific PR the debug plugin has been entirely rewritten so everything is a conflict anyhow :D hence merging this one)

Thanks!

**Pull request merged**